source/dub/dub.d

```diff
@@ -592,8 +592,17 @@ class Dub {
 			}
 		}
 
+		string[] missingDependenciesBeforeReinit = m_project.missingDependencies;
 		m_project.reinit();
 
+		if (!m_project.hasAllDependencies) {
+
+			string[] resolvedDependencies = setDifference(missingDependenciesBeforeReinit, m_project.missingDependencies).array;
```

Comment:
all you do is check `.empty` here, so why not store an auto with the setDifference?

```suggestion
			auto resolvedDependencies = setDifference(missingDependenciesBeforeReinit, m_project.missingDependencies);
```

Reply:
fixed